src/components/registration/index.js

```diff
@@ -24,11 +43,23 @@ const Registration = () => {
       <Row className='justify-content-center'>
         <Col xs={11} lg={6} className='my-2'>
           <Form onSubmit={onSubmit} className='needs-validation' noValidate>
-            <Email />
-            <Password />
-            <PasswordRepeat />
-            <Checkbox />
-            <FormButton />
+            <Email onTextChanged={onTextChanged('email')} state={state.email} />
+            <Password
+              onTextChanged={onTextChanged('password')}
+              state={state.password}
+            />
+            <PasswordRepeat
+              pattern={state.password.value}
+              onTextChanged={onTextChanged('passwordRepeat')}
+              state={state.passwordRepeat}
+              isValid={state.password.isValid}
+            />
+            <Checkbox
+              onClick={() =>
+                setState({ ...state, termsAccepted: !state.termsAccepted })
+              }
+            />
+            <FormButton disabled={!state.termsAccepted} />
```

Comment:
Tego chciałem uniknąć wprowadzając context, ale w sumie nie wygląda to aż tak źle 👍 .

Reply:
no niestety raczej ciężko tego uniknąć, context jest spoko, ale imo w tym przypadku tak jest dużo dużo czytelniej, a całość logiki zamyka się w jednym miejscu - i to nawet dość nieogromnym